python/_tskitmodule.c

```diff
@@ -1981,6 +2125,45 @@ EdgeTable_add_row(EdgeTable *self, PyObject *args, PyObject *kwds)
     return ret;
 }
 
+static PyObject *
+EdgeTable_update_row(EdgeTable *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *ret = NULL;
+    int err;
+    tsk_id_t row_index = -1;
+    double left = 0.0;
+    double right = 1.0;
+    int parent;
+    int child;
+    PyObject *py_metadata = Py_None;
+    char *metadata = "";
+    Py_ssize_t metadata_length = 0;
+    static char *kwlist[]
+        = { "row_index", "left", "right", "parent", "child", "metadata", NULL };
+
+    if (EdgeTable_check_state(self) != 0) {
+        goto out;
+    }
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O&ddii|O", kwlist, &tsk_id_converter,
+            &row_index, &left, &right, &parent, &child, &py_metadata)) {
+        goto out;
+    }
+    if (py_metadata != Py_None) {
+        if (PyBytes_AsStringAndSize(py_metadata, &metadata, &metadata_length) < 0) {
+            goto out;
+        }
+    }
+    err = tsk_edge_table_update_row(
```

Comment:
Should probably cast the ints to ``tsk_id_t``, and metadata_length to ``tsk_size_t``. (No need to go over old code, but for new code good to do it more correctly)

Reply:
Or, maybe we should just use your nice ``tsk_id_converter`` on all the tsk_id_t values, and save this annoying little dance?

Reply:
I've used `tsk_id_converter` and cast the string lengths where needed (including ancestral_state etc.)

Reply:
I've used `tsk_id_converter` and cast the string lengths where needed (including ancestral_state etc.)